SupportYourLocals.Data/CSVData.cs

```diff
@@ -5,73 +5,97 @@
+                {
+                    csvParser.CommentTokens = new[] { "#" };
+                    csvParser.SetDelimiters(new[] { "," });
+                    csvParser.HasFieldsEnclosedInQuotes = true;
 
-            var newLine = "{0},{1},{2}".Format(product, position, personsID);
+                    // Skip the row with the column names
+                    csvParser.ReadLine();
 
-            csv.AppendLine(newLine);
+                    while (!csvParser.EndOfData)
+                    {
+                        // Read current line fields, pointer moves to the next line.
+                        string[] fields = csvParser.ReadFields();
 
-            File.AppendAllText(filePath, csv.ToString());
+                        if (username == fields[0])
+                        {
+                            String salt = fields[2];
+                            if (CompareHash(password, Encoding.ASCII.GetBytes(fields[1]), salt))
+                            {
+                                return true;
+                            }
+                        }
+                    }
+                    return false;
+                }
+            }
         }
 
-        public static int GetPersonsID(int personsID)
+        public bool CheckRegisterData(String password, String username)
+        {
+            return false;
+        }
+        public void SaveRegisterData(String password, String username)
         {
-            // Checking person's ID
-            List<double> personID = new List<double>();
-            using (TextFieldParser csvParser = new TextFieldParser(filePath))
+            StringBuilder csv = new StringBuilder();
+
+            if (!File.Exists(filePath))
             {
-                csvParser.CommentTokens = new[] { "#" };
-                csvParser.SetDelimiters(new[] { "," });
-                csvParser.HasFieldsEnclosedInQuotes = true;
+                csv.AppendLine("Username, Password, ID");
+            }
 
-                // Skip the row with the column names
-                csvParser.ReadLine();
+            String salt = CreateSalt(saltSize);
 
-                while (!csvParser.EndOfData)
-                {
-                    // Read current line fields, pointer moves to the next line.
-                    string[] fields = csvParser.ReadFields();
-                    personID.Add(double.Parse(fields[3]));
-                }
-            }
+            //String hashedPassword = BitConverter.ToString(GenerateHash(password, salt)).Replace("-", "");
+
+            var newLine = "{0},{1},{2},{3}".Format(username, GenerateHash(password, salt), salt, personsID);
+
+            csv.AppendLine(newLine);
 
-            // Setting person's ID
-            personsID += personID.Count;
-            return personsID;
+            File.AppendAllText(filePath, csv.ToString());
         }
+        public bool CompareHash(string attemptedPassword, byte[] hash, string salt)
```

Comment:
Since this and the other password functions don't deal with class data, they can be made static.